api/threem.py

```diff
@@ -404,6 +401,19 @@ def process_one(self, error_tag, namespaces):
 
         if message in self.error_mapping:
             return self.error_mapping[message](message)
+        if message in ('Authentication failed', 'Unknown error'):
+            # 'Unknown error' is an unknown error on the 3M side.
+            #
+            # 'Authentication failed' could _in theory_ be an error on
+            # our side, but if authentication is set up improperly we
+            # actually get a 401 and no body. When we get a real error
+            # document with 'Authentication failed', it's always a
+            # transient error on the 3M side. Possibly some
+            # authentication internal to 3M has failed? Anyway, it
+            # happens relatively frequently.
+            return RemoteInitiatedServerError(
+                response.content.ThreeMAPI.SERVICE_NAME
```

Comment:
Aha! That code was raising an exception, but then the exception was being treated as an indicator of bad data from 3M, and a RemoteInitiatedServerError was created to represent the bad data. So my test was exercising this code and it was passing, but for totally the wrong reason.